docs/extensions/intro.md

```diff
@@ -0,0 +1,170 @@
+    --> Created tests/fix_testnet.py
+    --> Created tests/fix_keys_and_seeds.py
+    --> Created pyproject.toml
+    --> Created setup.py
+    --> Created setup.cfg
+    --> Created MANIFEST.in
+
+        Congratulations, you've created a new extension!
+
+        Here is how to get it to run in your development environment:
+            pip3 install -e .
+            python3 -m cryptoadvance.specter server --config DevelopmentConfig --debug
+            # Point your browser to http://localhost:25441
+            # Click "Choose plugins" --> rubberduck
+
+        If you want to package it, you can build it like this:
+            python3 -m pip install --upgrade build
+            python3 -m build
+            # Install it like this:
+            pip3 install dist/mynym_rubberduck-0.0.1-py3-none-any.whl
+
+        If you want to bring your extension to production, please refer to 
+        the readme in the dummy-extension repo:
+        https://github.com/cryptoadvance/specterext-dummy#how-to-get-this-to-production
+    
+        To publish your package
+
+            python3 -m pip install --upgrade twine
+            python3 -m twine upload --repository testpypi dist/*
+
+        You can get all this information again via:
+        python3 -m cryptoadvance.specter ext gen --help
+$ 
+```
+The created file structure looks like this and you will feel right at home if you have some knowledge about how Flask works:
+
+
+![](./images/extensions_file_layout.png)
+
+## Virtualenv Management
+
+The simplified example of creating an extension above is ommitting the use of a virtual environment. However, it's really __highly__ recommended that virtual environments are used for extension development. Without the Output above you would do something like:
+
+```
+mkdir /tmp/rubberduck && cd /tmp/rubberduck
+virtualenv --python=python3 .env
+. ./.env/bin/activate
+pip3 install cryptoadvance.specter 
+# you might want to have a specific version via:
+pip3 install cryptoadvance.specter==1.13.1
+python3 -m cryptoadvance.specter ext gen
+pip3 install -e .
+python3 -m cryptoadvance.specter server --config
+```
+
+Often enough, extension development goes hand in hand with the development of Specter Desktop iteself. Especially in the very beginning you need to constantly develop on the extension-concept on Specter Desktop while also using/consuming that functionality in an extension.
+
+In such a case, the best option is to "reuse" the virtualenv of Specter Desktop also for your extension. Other than what is described above, you would do something like the below. If you have checked out the Specter Desktop Code in parallel with the extension-code, you would ommit the creation of the virtualenv above and rather activate the existing one in your Specter Desktop Development Directory:
+
+```
+. ../specter-desktop/.env/bin/activate
+pip3 install -e .
+python3 -m cryptoadvance.specter server --config DevelopmentConfig --debug
+```
```

Comment:
Maybe it doesn't make a difference, but I think of it more as creating a virtualenv for the extension and then installing the Desktop core into my extension virtualenv, instead of the reverse. That way my extension development (including virtualenv) is totally self-contained -- if I delete the extension directory, I'm back to clean.

Reply:
I agree, it's just consequent. It's just that i didn't do that until now, but i'll change my practice here. I already suffered at least once because of unclear dependencies.
However, i've not yet tested that practice yet. So I've added it as the better approach but not yet tested.